### ISSUE TITLE
Synthizer crashes when files end playing

### ISSUE THREAD

**Issue by chrisnorman7** [synthizer/synthizer#13]
OK, so here's my latest bug.

There are some files that synthizer doesn't like it would seem. I have two name.wav files in my lucky_thirteen game, and one of them (the one generated by gtts then converted by goldwave) causes the crash.

Here is Synthizer's output:

```
debug(context-thread 1) Thread started
debug(unknown-thread 2) Handling path sounds/voices/google/name.wav with handler dr_wav
```

Here is the code I used to test:

```
from time import sleep
import synthizer

with synthizer.initialized():
    c = synthizer.Context()
    synthizer.configure_logging_backend(synthizer.LoggingBackend.STDERR)
    synthizer.set_log_level(synthizer.LogLevel.DEBUG)
    b = synthizer.Buffer.from_stream('file', 'sounds/voices/google/name.wav')
    g = synthizer.BufferGenerator(c)
    g.buffer = b
    s = synthizer.DirectSource(c)
    s.add_generator(g)
    sleep(4)
```

And here is the result of ffprobe -i sounds/voices/google/name.wav:

```
ffprobe version 4.1.6-1~deb10u1 Copyright (c) 2007-2020 the FFmpeg developers
  built with gcc 8 (Debian 8.3.0-6)
  configuration: --prefix=/usr --extra-version='1~deb10u1' --toolchain=hardened --libdir=/usr/lib/x86_64-linux-gnu --incdir=/usr/include/x86_64-linux-gnu --arch=amd64 -
-enable-gpl --disable-stripping --enable-avresample --disable-filter=resample --enable-avisynth --enable-gnutls --enable-ladspa --enable-libaom --enable-libass --enable
-libbluray --enable-libbs2b --enable-libcaca --enable-libcdio --enable-libcodec2 --enable-libflite --enable-libfontconfig --enable-libfreetype --enable-libfribidi --ena
ble-libgme --enable-libgsm --enable-libjack --enable-libmp3lame --enable-libmysofa --enable-libopenjpeg --enable-libopenmpt --enable-libopus --enable-libpulse --enable-
librsvg --enable-librubberband --enable-libshine --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libssh --enable-libtheora --enable-libtwolame --enable-l
ibvidstab --enable-libvorbis --enable-libvpx --enable-libwavpack --enable-libwebp --enable-libx265 --enable-libxml2 --enable-libxvid --enable-libzmq --enable-libzvbi --
enable-lv2 --enable-omx --enable-openal --enable-opengl --enable-sdl2 --enable-libdc1394 --enable-libdrm --enable-libiec61883 --enable-chromaprint --enable-frei0r --ena
ble-libx264 --enable-shared
  libavutil      56. 22.100 / 56. 22.100
  libavcodec     58. 35.100 / 58. 35.100
  libavformat    58. 20.100 / 58. 20.100
  libavdevice    58.  5.100 / 58.  5.100
  libavfilter     7. 40.101 /  7. 40.101
  libavresample   4.  0.  0 /  4.  0.  0
  libswscale      5.  3.100 /  5.  3.100
  libswresample   3.  3.100 /  3.  3.100
  libpostproc    55.  3.100 / 55.  3.100
Input #0, wav, from 'name.wav':
  Duration: 00:00:00.82, bitrate: 768 kb/s
    Stream #0:0: Audio: pcm_s16le ([1][0][0][0] / 0x0001), 24000 Hz, 2 channels, s16, 768 kb/s
```

If I change the code to run with sounds/voices/chris/name.wav, I get the following Synthizer output:

```
debug(context-thread 1) Thread started
debug(unknown-thread 2) Handling path sounds/voices/chris/name.wav with handler dr_wav
debug(background-thread 3) Context shutdown
debug(context-thread 1) Context thread terminating
debug(background-thread 3) Background thread stopped
debug(unknown-thread 4) Deferred free processed 6 frees in a background thread
debug(unknown-thread 2) Library shutdown complete
```

Here is the result of ffprobe on that file:

```
ffprobe version 4.1.6-1~deb10u1 Copyright (c) 2007-2020 the FFmpeg developers
  built with gcc 8 (Debian 8.3.0-6)
  configuration: --prefix=/usr --extra-version='1~deb10u1' --toolchain=hardened --libdir=/usr/lib/x86_64-linux-gnu --incdir=/usr/include/x86_64-linux-gnu --arch=amd64 -
-enable-gpl --disable-stripping --enable-avresample --disable-filter=resample --enable-avisynth --enable-gnutls --enable-ladspa --enable-libaom --enable-libass --enable
-libbluray --enable-libbs2b --enable-libcaca --enable-libcdio --enable-libcodec2 --enable-libflite --enable-libfontconfig --enable-libfreetype --enable-libfribidi --ena
ble-libgme --enable-libgsm --enable-libjack --enable-libmp3lame --enable-libmysofa --enable-libopenjpeg --enable-libopenmpt --enable-libopus --enable-libpulse --enable-
librsvg --enable-librubberband --enable-libshine --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libssh --enable-libtheora --enable-libtwolame --enable-l
ibvidstab --enable-libvorbis --enable-libvpx --enable-libwavpack --enable-libwebp --enable-libx265 --enable-libxml2 --enable-libxvid --enable-libzmq --enable-libzvbi --
enable-lv2 --enable-omx --enable-openal --enable-opengl --enable-sdl2 --enable-libdc1394 --enable-libdrm --enable-libiec61883 --enable-chromaprint --enable-frei0r --ena
ble-libx264 --enable-shared
  libavutil      56. 22.100 / 56. 22.100
  libavcodec     58. 35.100 / 58. 35.100
  libavformat    58. 20.100 / 58. 20.100
  libavdevice    58.  5.100 / 58.  5.100
  libavfilter     7. 40.101 /  7. 40.101
  libavresample   4.  0.  0 /  4.  0.  0
  libswscale      5.  3.100 /  5.  3.100
  libswresample   3.  3.100 /  3.  3.100
  libpostproc    55.  3.100 / 55.  3.100
Input #0, wav, from '/mnt/c/Users/chris/src/lucky_thirteen/sounds/voices/chris/name.wav':
  Duration: 00:00:00.63, bitrate: 1411 kb/s
    Stream #0:0: Audio: pcm_s16le ([1][0][0][0] / 0x0001), 44100 Hz, 2 channels, s16, 1411 kb/s
```

If you need any more information, please let me know.

**Comment by camlorn** [comment]
I probably won't get to this until next weekend. These files can be pulled from your project?

**Comment by chrisnorman7** [comment]
Yes. They're in the sounds/voices subdirectory.

Most - if not all - of the ones in the Google directory elicit the same
result.

Take care,

Chris Norman

**Comment by camlorn** [comment]
This should now be fixed and in 0.6.3.

**Comment by chrisnorman7** [comment]
Perfect, thank you.
